File: dsl/src/main/java/org/nsu/dsl/block/SingletonSemaphore.java
package org.nsu.dsl.block;

import java.util.concurrent.Semaphore;

public class SingletonSemaphore {

    private static Semaphore semaphore;

    public static void setLimit(int limit){
        semaphore = new Semaphore(limit);
    }

    public static Semaphore getSemaphore() {
        return semaphore;
    }
}

File: dsl/src/main/kotlin/org/nsu/dsl/semaphore/SingletonSemaphore.kt
package org.nsu.dsl.semaphore

import java.util.concurrent.Semaphore

class SingletonSemaphore {
    companion object {

        private var semaphore: Semaphore? = null

        fun setLimit(limit: Int = 5) {
            semaphore = Semaphore(limit)
        }

        fun getSemaphore(): Semaphore = semaphore!!
    }
}
